### PR TITLE
Feature/Database and Authentication

### PR DESCRIPTION
**Pull request by augisbud**
Install Docker Desktop

`docker run -d --name postgres -p 5432:5432 -e POSTGRES_PASSWORD=supersecret postgres`

`dotnet ef database update`

### CODE REVIEW

**Review by Edvinas11 — CHANGES_REQUESTED** [review]
Need to do CORS settings

File: InvoicesBackend/Program.cs [hunk 1]
```diff
@@ -1,19 +1,86 @@
...
+    var xmlFile = $"{Assembly.GetExecutingAssembly().GetName().Name}.xml";
+    var xmlPath = Path.Combine(AppContext.BaseDirectory, xmlFile);
+    options.IncludeXmlComments(xmlPath);
+});
+
+builder.Services.AddDbContext<AppDbContext>(options => options.UseNpgsql(builder.Configuration.GetValue<string>("ConnectionStrings:DefaultConnection")));
+
+builder.Services.AddAutoMapper(AppDomain.CurrentDomain.GetAssemblies());
+
+builder.Services.AddScoped<InvoicesRepository>();
```
Comment (Edvinas11) [inline InvoicesBackend/Program.cs:81]: You can add these lines here
`builder.Services.AddCors(options => {
    options.AddPolicy("AllowSpecificOrigin", builder => {
        builder.AllowAnyOrigin();
        builder.AllowAnyMethod();
        builder.AllowAnyHeader();
    });
});`

File: InvoicesBackend/Program.cs [hunk 2]
```diff
@@ -1,19 +1,86 @@
...
 
 var app = builder.Build();
 
-if (app.Environment.IsDevelopment())
-{
-    app.UseSwagger();
-    app.UseSwaggerUI();
-}
+app.UseSwagger();
+app.UseSwaggerUI();
```
Comment (Edvinas11) [inline InvoicesBackend/Program.cs:87]: Add additional line here: `app.UseCors("AllowSpecificOrigin");`

**Review by Edvinas11 — CHANGES_REQUESTED** [review]

File: InvoicesBackend/Controllers/LoginController.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+                _config["Jwt:Issuer"],
+                _config["Jwt:Issuer"],
+                null,
+                expires: DateTime.Now.AddMinutes(60),
+                signingCredentials: credentials
+            );
+
+            var token =  new JwtSecurityTokenHandler().WriteToken(Sectoken);
+
+            return Ok(token);
```
Comment (Edvinas11) [inline InvoicesBackend/Controllers/LoginController.cs]: Should return a JSON format token `return Ok(new { token });`

Comment (augisbud) [reply InvoicesBackend/Controllers/LoginController.cs]: Addressed in https://github.com/augisbud/PROPRO/pull/7/commits/8eda1b360955d00f8a7b355ce4aa394c12489b26

File: InvoicesBackend/Program.cs (hunk 2 above)
Comment (augisbud) [reply InvoicesBackend/Program.cs:87]: Addressed in https://github.com/augisbud/PROPRO/pull/7/commits/8eda1b360955d00f8a7b355ce4aa394c12489b26

File: InvoicesBackend/Program.cs (hunk 1 above)
Comment (augisbud) [reply InvoicesBackend/Program.cs:81]: Addressed in https://github.com/augisbud/PROPRO/pull/7/commits/8eda1b360955d00f8a7b355ce4aa394c12489b26

**Review by Edvinas11 — APPROVED** [review]

**Pull request merged**